### PR TITLE
zstd: add pthread to FreeBSD too

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **zstd/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zstd/all/conanfile.py [hunk 1]
```diff
@@ -72,5 +72,5 @@ def package_info(self):
         self.cpp_info.components["zstdlib"].names["cmake_find_package"] = zstd_cmake
         self.cpp_info.components["zstdlib"].names["cmake_find_package_multi"] = zstd_cmake
         self.cpp_info.components["zstdlib"].libs = tools.collect_libs(self)
-        if self.settings.os == "Linux":
+        if self.settings.os == "Linux" or self.settings.os == "FreeBSD":
```
Comment (SpaceIm) [inline recipes/zstd/all/conanfile.py:75]:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
```

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@SpaceIm are you ok with this PR ?

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
Does it mean that all CI recipes with pthread dependency should be updated accordingly?

**Comment by ericLemanissier** [comment]
it's probable. I just happened to stumble on zstd and sqlite3 in @patricia-gallardo's tests at https://github.com/turtlebrowser/turtlebrowser/runs/1532253993?check_suite_focus=true#step:3:14003 for example

**Review by uilianries — APPROVED** [review]

**Comment by SSE4** [comment]
yes, probably, it has to be fixed for more recipes. and probably, It needs to be addressed for more operation systems (e.g. NetBSD, OpenBSD, etc.).

**Pull request merged**